src/pages/in/embed/nft/class/_classId.vue

```diff
@@ -0,0 +1,190 @@
+<template>
+  <NFTWidget
+    ref="widget"
+    :title="contentTitle"
+    :description="contentDescription"
+    :img-src="contentImage"
+    :url="contentURL"
+    :price="nftPrice"
+    :collector-count="nftCollectorCount"
+    :collected-count="nftCollectedCount"
+    :owner-address="iscnOwnerAddress"
+    :owner-display-name="iscnOwnerDisplayName"
+    :owner-avatar-src="iscnOwnerAvatarSrc"
+    :is-content-clickable="false"
+    @view-details="viewNFTDetails"
+    @collect="collectNFT"
+    @like="likeISCN"
+    @load-image="notifyParentOfResizing"
+  />
+</template>
+
+<script>
+import {
+  APP_LIKE_CO_URL_BASE,
+  LIKER_LAND_URL_BASE,
+} from '~/constant';
+import {
+  apiGetLikerDataByAddress,
+  apiGetNFTMetadata,
+  apiGetNFTOwners,
+  apiGetNFTPurchaseInfo,
+} from '~/util/api/api';
+import { getClassInfo } from '~/util/nft';
+
+export default {
+  layout: 'widget',
+  computed: {
+    widgetId() {
+      return this.$route.query.wid;
+    },
+    detailsURL() {
+      return `${LIKER_LAND_URL_BASE}/nft/class/${this.classId}`;
+    },
+    purchaseURL() {
+      return `${APP_LIKE_CO_URL_BASE}/nft/purchase/${encodeURIComponent(
+        this.iscnId
+      )}`;
+    },
+  },
+  async asyncData({ params }) {
+    const { classId } = params;
+    let metadata;
+    const [
+      chainMetadata = {},
+      apiMetadataResult,
+      ownersResult,
+      purchaseInfoResult,
+    ] = await Promise.all([
+      getClassInfo(classId),
+      apiGetNFTMetadata({ classId })
+        // eslint-disable-next-line no-console
+        .catch(error => console.error(error)),
```

Comment:
Just as a note: if we want to support NFT other than writing NFT in the future, metadata should be fetched from `uri` field of the on-chain metadata